ethereum/contracts/SwapFactory.sol

```diff
@@ -61,11 +60,14 @@ contract SwapFactory is ERC2771Context, Secp256k1 {
 
     // newSwap creates a new Swap instance with the given parameters.
     // it returns the swap's ID.
+    // _timeoutDuration0: duration between the current timestamp and timeout0
+    // _timeoutDuration1: duration between timeout0 and timeout1
     function newSwap(
         bytes32 _pubKeyClaim,
         bytes32 _pubKeyRefund,
         address payable _claimer,
-        uint256 _timeoutDuration,
+        uint256 _timeoutDuration0,
+        uint256 _timeoutDuration1,
```

Comment:
At some point we discussed setting these timeout durations at contract creation time instead of doing it on every single new swap. Then `newSwap` transactions would use slightly less gas, but we'd still be able to optimize each contract deployment for the block interval of the deployed chain. Would there be any downside in doing this?

Reply:
I can check the gas savings if we do that though, might be worth it if it's significant

Reply:
I tried it out and it only saved 1387 gas units (went from 52594 to 51207). The test I ran moved the timeout values to the constructor and set them on immutable variables, then I removed them from the `newSwap` arguments and removed them from the emitted logs, since you can calculate their values based off of the block timestamp that the newSwap transaction was mined into.

If the timeout values change, we'd have to update the protocol version to prevent failed swaps, and if the protocol version is being changed, redeploying might not be that big of a deal. But I agree, if it is not saving an amount of gas that anyone cares about, flexibility is better.